options/glibc/generic/search.cpp

```diff
@@ -0,0 +1,17 @@
+#include <bits/glibc/glibc_search.h>
+#include <bits/ensure.h>
+
+int hsearch_r(ENTRY, ACTION, ENTRY **, struct hsearch_data *) {
+	__ensure(!"Not implemented");
+	__builtin_unreachable();
+}
+
+int hcreate_r(size_t, struct hsearch_data *) {
+	__ensure(!"Not implemented");
+	__builtin_unreachable();
+}
+
+void hdestroy_r(struct hsearch_data *) {
+	__ensure(!"Not implemented");
+	__builtin_unreachable();
+}
```

Comment:
Why not actually implement these? Also perhaps the actual implementations should be shared (in `options/internal`?), since AFAICS the non-reentrant versions are just wrappers around these, except that they pass a pointer to some global instance of `struct hsearch_data`.

I propose making out `struct hsearch_data` only contain one pointer to an internal heap-allocated object, which would contain perhaps a `frg::hash_map`? That way we can also extend the struct as needed without ABI breakage.

Reply:
Yeah thats a good point. Ill do that.